src/spec/core.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <array>
+#include <string>
+#include <string_view>
+
+namespace AQNWB::spec::core
```

Comment:
For consistency, I think it would be nicer to use all uppercase for the namespaces:

```suggestion
namespace AQNWB::SPEC::CORE
```